flow/project.py

```diff
@@ -3003,8 +3294,25 @@ def submit(self, bundle_size=1, jobs=None, names=None, num=None, parallel=False,
             :py:class:`~.IgnoreConditions`
         """
         # Regular argument checks and expansion
-        if jobs is None:
-            jobs = self  # select all jobs
+        if jobs is not None:
+            aggregates = set()  # Set in order to prevent duplicate entries
+            for job in jobs:
+                if isinstance(job, signac.contrib.job.Job):  # User can still pass signac jobs
+                    aggregates.add((job,))
+                else:
+                    try:
+                        aggregates.add(tuple(job))  # An aggregate provided by the user
+                    except Exception:
+                        raise ValueError('Invalid aggregate provided by the user. If you want to '
+                                         'provide an aggregate, please pass a tuple of jobs '
+                                         'instead.')
+        else:
+            aggregates = None
+
+        # Even though we create the aggregates from all the jobs in a FlowProject,
+        # a user can still pass in jobs / aggregates which are invalid.
+        # Hence, we need to verify them.
+        self._verify_aggregate_project(aggregates)
```

Comment:
We should probably verify when iterating over the passed jobs. Since this is likely to be needed multiple places, we could make it a function that returns the correct list or yields valid aggregates.